xarray/core/dataset.py

```diff
@@ -2065,15 +2084,18 @@ def tail(
         return self.isel(indexers)
 
     def thin(
-        self, indexers: Mapping[Hashable, Any] = None, **indexers_kwargs: Any
+        self,
+        indexers: Union[Mapping[Hashable, Any], int] = None,
```

Comment:
Is this right? 

```suggestion
        indexers: Union[Mapping[Hashable, int], int] = None,
```

Reply:
I'm kind of new to this typing thing but it seems logical to me. Thanks for the suggestion!